File: src/SampleWebsite.Models/Hubs/PersonHub.cs
using Microsoft.AspNet.SignalR;

namespace SampleWebsite.Models.Hubs
{
    public class PersonHub : Hub<IPersonHubClient>
    {
        public void CreatePerson(Person person)
        {
            // Save the person or whatever
            // Notify other users that a new person has been created
            Clients.All.PersonCreated(person);
        }
    }

    public interface IPersonHubClient
    {
        /// <summary>
        /// Notify clients when a new person has been created
        /// </summary>
        void PersonCreated(Person person);
    }
}

File: src/SampleWebsite/Scripts/App/PersonHub.ts
class PersonHub {

    private static initialized: boolean = false;

    constructor() {


        if (!PersonHub.initialized) {
            const hub = $.connection.personHub;

            // Implement all the client-side callbacks defined on IPersonHubClient (in hubs.d.ts)
            hub.client.personCreated = p => {
                console.log("A person has been created [Name = " + p.Name + ", Gender = " + p.Gender + "]");
            };

            PersonHub.initialized = true;
        }

    }

}
